### ISSUE TITLE
Standardise the usage of sqlalchemy engine across the package

### ISSUE THREAD

**Issue by deniztepe** [OpenEnergyPlatform/open-MaStR#267]
While skimming through sqlalchemy documentation, I have realised a few issues about how we implement the database connection and the transactions with the help of sqlalchemy.engine.Engine object. I think the code can be written with less repetition and higher performance.

- [x] Ambiguous naming of attributes. DB_ENGINE refers to the database management system name such as sqlite, whereas _engine the real engine object is. 
`open_mastr/mastr.py:53`
```
        # setup database engine and connection
        self.DB_ENGINE = os.environ.get("DB_ENGINE", "sqlite")
        self._engine = db_engine()
```
- [x] Inside the helper function `session_scope` in `open_mastr.utils.helpers.session_scope`, a new engine is created at every function call. Each transaction leads to a new connection. Instead, the object attribute self._engine could be engaged. 

I would edit the issue, if I find further points. Please remark if there is a particular reason for any of the implementations. 

I would let following considerations for further issues

- Enable the selection of database system during the instantiation of mastr object. Input types like "None, 'sqlite', 'docker', sqlalchemy.engine.Engine". The latest for an existing database. As default fallback, the environment variables can  be used. 
- Switch to sqlalchemy dialect where sqlite3 has been used.

**Comment by FlorianK13** [comment]
At the initialization of a Mastr object a new parameter `engine` now exists. This parameter can take values 'sqlite', 'docker-postgres', or a sqlalchemy.engine.Engine object.

**Comment by deniztepe** [comment]
Merged by #270
